### PR TITLE
SDSS-363: Replaced dek with dek long in news list view pattern.

### PR DESCRIPTION
**Pull request by joegl**
# READY FOR REVIEW

# Summary
- Replaced dek with dek long in news list view pattern.
- Updated the display to have the long dek.
- Moved the order of date and source in the News list and teasers 
- Add the long dek  to the News list and teasers 

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Urgency
- How urgent is this? (Normal, High)

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Rebuild Cache and import config `drush cr ; drush ci`
3. Navigate to http://{Localbuild}/news/school-highlights
4. Verify the list shows external link icons and the links work.
5. Create a page with teasers and a list and verify the order and display per the design in [SDSS-381](https://stanfordits.atlassian.net/browse/SDSS-381)

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory?

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- [https://stanfordits.atlassian.net/browse/SDSS-381](https://stanfordits.atlassian.net/browse/SDSS-381)
- [https://stanfordits.atlassian.net/browse/SDSS-363](https://stanfordits.atlassian.net/browse/SDSS-363)
- [https://stanfordits.atlassian.net/browse/SDSS-412](https://stanfordits.atlassian.net/browse/SDSS-412)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

**Comment by jenbreese** [comment]
I worked on styling and adding the following to this pr. 
SDSS-412 - styling for the new Long Dek
SDSS-381  - Styling for the news teaser and list because these files are all intertwined.

**Comment by joegl** [comment]
@jenbreese When I add a value to a News Source field and try to load the QA-List page, I get a 500 error. I'm trying to debug it locally, but I wanted to know if you had any ideas/experience the same thing.

**Comment by jenbreese** [comment]
@joegl I don't recall that problem. I will switch branches and look.

**Comment by jenbreese** [comment]
@joegl I'm not able to replicate what a 500 error. I can load the QA pages, edit content to add a source and then see that on the page when I save it.

File: themes/sdss_subtheme/templates/components/news-list/news-item-list.html.twig [hunk 1]
```diff
@@ -68,17 +79,26 @@
...
+        </div>
+
+
+        {# News List External link  #}
+        <div class="news-vertical-teaser__ext-link">{{ su_news_source }}</div>
+
+        {# News List Long Dek #}
+        {# Since CSS Grid is used, always render an element even if it's empty. #}
+        <div class="su-news-list__long-dek">
+          {% if news_list_publishing_date|render|striptags('<drupal-render-placeholder>')|trim is not empty %}
```
Comment (joegl) [inline themes/sdss_subtheme/templates/components/news-list/news-item-list.html.twig]: Should this be checking the value of the long dek field and not the publishing date?

File: themes/sdss_subtheme/templates/components/news-vertical-teaser/pattern-news-vertical-teaser.html.twig [hunk 2]
```diff
@@ -61,6 +61,9 @@
     {%- if news_url is not empty -%}
       </a>
     {%- endif -%}
+      <div class="news-vertical-teaser__source">{{ news_source }}</div>
+      <div class="news-vertical-teaser__ext-link">{{su_news_source}}</div>
```
Comment (joegl) [inline themes/sdss_subtheme/templates/components/news-vertical-teaser/pattern-news-vertical-teaser.html.twig]: What are both of these doing in here?

File: themes/sdss_subtheme/templates/components/news-list/news-list.ui_patterns.yml [hunk 3]
```diff
@@ -31,12 +36,17 @@ sdss-news-list:
...
     news_url:
       type: text
       label: "URL for item"
       description: "URL for image and heading"
       preview: "http://www.stanford.edu/"
+    su_news_source:
+      type: text
+      label: "External URL for item"
+      description: "URL for image and heading"
+      preview: "http://www.google.com/"
```
Comment (joegl) [inline themes/sdss_subtheme/templates/components/news-list/news-list.ui_patterns.yml]: `su_news_source` gets confusing fast when working in the templates. Would we want to rename this one to `news_external_url`? That would follow the conventions in the rest of the component and be easier to understand.

File: themes/sdss_subtheme/templates/components/news-list/news-list.ui_patterns.yml [hunk 4]
```diff
@@ -22,6 +22,11 @@ sdss-news-list:
       label: "Dek"
       description: "This is the dek specific to news items."
       preview: "<p>Sample text goes here and this list should be able to handle a few lines.</p>"
+    news_list_long_dek:
+      type: text
+      label: "Long Dek"
+      description: "This is the long dek specific to news items."
+      preview: "<p>Sample text goes here and this list should be able to handle a few lines.</p>"
```
Comment (joegl) [inline themes/sdss_subtheme/templates/components/news-list/news-list.ui_patterns.yml:29]: Do we need to add a new long dek field to the component? The Dek (long) field was just to allow them to use a longer dek in place of the current dek field, so we can just use the existing dek field in the component.

File: themes/sdss_subtheme/templates/components/news-vertical-teaser/pattern-news-vertical-teaser.ui_patterns.yml [hunk 5]
```diff
@@ -32,6 +37,11 @@ sdss-news-vertical-teaser:
       label: "URL for item"
       description: "URL for image and heading"
       preview: "http://www.stanford.edu/"
+    su_news_source:
+      type: text
+      label: "External URL for item"
+      description: "URL for image and heading"
+      preview: "http://www.google.com/"
```
Comment (joegl) [inline themes/sdss_subtheme/templates/components/news-vertical-teaser/pattern-news-vertical-teaser.ui_patterns.yml]: Same question if we'd want to rename `su_news_source` to `news_external_url`

**Review by joegl — COMMENTED** [review]
This is looking good! I left a few comments to address and then just had one more question:

* The design in the ticket for the Source field has a "Source" label (e.g., Source: Los Angeles Times). Is it supposed to be there or was leaving it off intentional?

File: themes/sdss_subtheme/templates/components/news-list/news-item-list.html.twig (hunk 1 above)
Comment (jenbreese) [reply themes/sdss_subtheme/templates/components/news-list/news-item-list.html.twig]: The Missing label is a mistake.

File: themes/sdss_subtheme/templates/components/news-vertical-teaser/pattern-news-vertical-teaser.html.twig (hunk 2 above)
Comment (jenbreese) [reply themes/sdss_subtheme/templates/components/news-vertical-teaser/pattern-news-vertical-teaser.html.twig]: That is a mistake.

File: themes/sdss_subtheme/templates/components/news-list/news-list.ui_patterns.yml (hunk 4 above)
Comment (jenbreese) [reply themes/sdss_subtheme/templates/components/news-list/news-list.ui_patterns.yml:29]: I had thought we'd want to use the longer dek so it was less confusing to later devs.

Comment (joegl) [reply themes/sdss_subtheme/templates/components/news-list/news-list.ui_patterns.yml:29]: The name is fine. But if we are only ever going to display one dek field, the component should only have one dek field. Then the site builder chooses in the "Manage Display" tab for the component which field from the News content type (original dek field or dek (long) field) to use for the dek field display in the component. I hope that makes sense!

File: themes/sdss_subtheme/templates/components/news-vertical-teaser/pattern-news-vertical-teaser.html.twig (hunk 2 above)
Comment (jenbreese) [reply themes/sdss_subtheme/templates/components/news-vertical-teaser/pattern-news-vertical-teaser.html.twig]: I fixed this in the new push.

File: themes/sdss_subtheme/templates/components/news-list/news-item-list.html.twig (hunk 1 above)
Comment (jenbreese) [reply themes/sdss_subtheme/templates/components/news-list/news-item-list.html.twig]: I fixed this in the new push.

File: themes/sdss_subtheme/templates/components/news-list/news-list.ui_patterns.yml (hunk 4 above)
Comment (jenbreese) [reply themes/sdss_subtheme/templates/components/news-list/news-list.ui_patterns.yml:29]: I'm not exactly sure what you would like on this file.  Should I remove the long dek from the patterns.yml file?

File: themes/sdss_subtheme/templates/components/news-vertical-teaser/pattern-news-vertical-teaser.ui_patterns.yml (hunk 5 above)
Comment (jenbreese) [reply themes/sdss_subtheme/templates/components/news-vertical-teaser/pattern-news-vertical-teaser.ui_patterns.yml]: Ok, all done.

**Review by joegl — COMMENTED** [review]
This is all working and acting for me as expected. Thanks Jen!!

**Pull request merged**